presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -83,7 +83,7 @@ PrestoExchangeSource::PrestoExchangeSource(
       clientCertAndKeyPath_(clientCertAndKeyPath),
       ciphers_(ciphers),
       exchangeExecutor_(executor) {
-  folly::SocketAddress address(folly::IPAddress(host_).str(), port_, true);
+  folly::SocketAddress address(host_, port_, true);
```

Comment:
why do we need this change?

Reply:
When we give hostname in `node.internal-address`, this line will try to resolve it as IP address and fail. Had to correct that. The third argument which was already set to true will make sure name look up happens.

Reply:
No, it won't

Reply:
@nmahadevuni The error happens here in SocketAddress

https://github.com/facebook/folly/blob/main/folly/SocketAddress.cpp#L632-L639

The resolution fails is ip address is passed instead of hostname and allowNameLookup is true. It may not always fail, and depends a lot on the host set up I think. Regardless, the look up is slow, lets keep the fast path with just ip address and use internal-address as hostname and create different constructor for exchange that only takes name (not  IP). Also instead of internal-address, shall we name hostname? That will be even more clear.

Reply:
internal-address is the config name in Presto, so wanted to have the same name in Native worker too.